dev_guide/templates.adoc

```diff
@@ -172,6 +193,48 @@ the following command:
 $ oc edit template <template>
 ----
 
+=== Defining Template parameters
+
+Each parameter describes a variable and the variable value which can be
```

Comment:
needs an intro sentence.  "Templates allow you to define parameters which take on a value.  That value is then substituted whereever the parameter is referenced".  References can be defined in any text field in the `objects` list field.  The value can be set explicitly or it can be generated by OpenShift.